### PR TITLE
[TR] PIM-2735: Allow to register custom mass edit JS actions

### PR DESCRIPTION
**Pull request by grena**
| Q                 | A
| ----------------- | ---
| Added Specs       | N
| Added Behats      | Y
| Travis CI is ok   | Y
| Changelog updated | N

### CODE REVIEW

**Comment by ClementGautier** [comment]
GTM

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid-builder.js [hunk 1]
```diff
@@ -170,7 +178,11 @@ define(function (require) {
 
                 // mass actions
                 _.each(metadata.massActions, function (options, action) {
-                    massActions[action] = modules[helpers.actionType('mass')].extend(options);
+                    if (!_.contains(reservedActions, options.type)) {
+                        massActions[action] = modules[helpers.actionType(options.type)].extend(options);
+                    } else {
+                        massActions[action] = modules[helpers.actionType('mass')].extend(options);
+                    }
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid-builder.js]:
```
var actionType = _.contains(reservedActions, options.type) ? 'mass' : options.type;
massActions[action] = modules[helpers.actionType(actionType)].extend(options);
```
?

**Comment by phaseinducer** [comment]
GTM

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid-builder.js (hunk 1 above)
Comment (grena) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid-builder.js]: yes, nice one! thanks

**Pull request merged**